src/wallet/salvage.cpp

```diff
@@ -18,6 +18,13 @@ typedef std::pair<std::vector<unsigned char>, std::vector<unsigned char> > KeyVa
 
 bool RecoverDatabaseFile(const fs::path& file_path, bilingual_str& error, std::vector<bilingual_str>& warnings)
 {
+    DatabaseOptions options;
+    DatabaseStatus status;
+    options.require_existing = true;
+    options.verify = false;
+    std::unique_ptr<WalletDatabase> database = MakeDatabase(file_path, options, status, error);
+    if (!database) return false;
```

Comment:
I think we need to close and destroy the `database`. Having it created and potentially open now could interfere with the salvage below. We only want to do this for the path existence check.

Reply:
What I was asking for was to have a `database = nullptr` here so that `database` isn't valid further down.

If we do want to do the salvage through `WalletDatabase` (I'm not sure that we should though), then I would expect that all of this recovery code ends up back in `bdb.cpp/h`.